### PR TITLE
Fix service starting for Ubuntu 16.04 (xenial)

### PR DESCRIPTION
**Pull request by gigo6000**
This fixes a bug when starting the service on Ubuntu 16.04 (xenial).

More details of the bug here: https://github.com/sous-chefs/mongodb/issues/132

### CODE REVIEW

File: definitions/mongodb.rb [hunk 1]
```diff
@@ -195,7 +195,14 @@
...
   # service
   service new_resource.name do
-    provider Chef::Provider::Service::Upstart if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
+    case node['platform']
+    when 'ubuntu'
+      if node['platform_version'] == '16.04'
+        provider Chef::Provider::Service::Systemd
+      end
+    else
+      provider Chef::Provider::Service::Upstart if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
```
Comment (shortdudey123) [inline definitions/mongodb.rb]: This should be an `elsif` on the `if` statement, not `case`

File: definitions/mongodb.rb [hunk 2]
```diff
@@ -195,7 +195,14 @@
 
   # service
   service new_resource.name do
-    provider Chef::Provider::Service::Upstart if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
+    case node['platform']
+    when 'ubuntu'
+      if node['platform_version'] == '16.04'
```
Comment (shortdudey123) [inline definitions/mongodb.rb]: `>= '15.04'`

File: definitions/mongodb.rb (hunk 1 above)
Comment (gigo6000) [inline definitions/mongodb.rb]: I used a `case` so more versions could be  added later and is easier to read than a bunch of `if`s

Comment (gigo6000) [inline definitions/mongodb.rb]: BTW I used the resource documentation as a reference: https://docs.chef.io/resource_service.html

**Comment by gigo6000** [comment]
I think is ready to be merged now.

File: definitions/mongodb.rb (hunk 1 above)
Comment (shortdudey123) [inline definitions/mongodb.rb]: This resource block will be removed within the next week so its probably a moot point anyway

**Pull request merged**